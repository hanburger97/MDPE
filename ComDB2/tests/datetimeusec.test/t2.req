set timezone UTC
insert into dtus(id, dt) values(1, 3600.246)
insert into dtus(id, dt) values(2, 3600.246246)
insert into dtus(id, dt) values(3, "7200.246")
insert into dtus(id, dt) values(4, cast("2015-02-23T080000.246" as datetime))
insert into dtus(id, dt) values(5, "2015-02-23T080000.246")
insert into dtus(id, dt) values(5, "2015-02-23T080000.000246")
insert into dtus(id, dt) values(6, "2015-02-23T080000.246246")
select cast(dt as TEXT), id from dtus order by dt, id
select cast(dt as TEXT), id from dtus where dt > 10000 order by id
set timezone US/Eastern
select cast(dt as TEXT), id from dtus order by dt, id
insert into dtus(id, ds) values(7, 3600.246)
insert into dtus(id, ds) values(8, 3600.246246)
insert into dtus(id, ds) values(9, "7200.246")
insert into dtus(id, ds) values(10, cast(1.246 as second))
insert into dtus(id, ds) values(11, cast(1.246246 as second))
insert into dtus(id, ds) values(12, 10)
insert into dtus(id, ds) values(13, "10 00:00:00.123456")
insert into dtus(id, ds) values(14, "0 02:04:06.246")
select cast(ds as TEXT), id from dtus order by ds, id
select cast(ds as TEXT), id from dtus where ds > "0 01:00:00" order by ds, id
select cast(dt as TEXT), cast(ds as TEXT), id from dtus order by id
select count(*) from dtus where dt < "2045-02-23T080000.246 UTC"
