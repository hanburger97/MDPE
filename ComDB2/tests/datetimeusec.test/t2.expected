(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(cast(dt as TEXT)='1970-01-01T010000.246000 UTC', id=1)
(cast(dt as TEXT)='1970-01-01T010000.246246 UTC', id=2)
(cast(dt as TEXT)='1970-01-01T020000.246000 UTC', id=3)
(cast(dt as TEXT)='2015-02-23T080000.000246 UTC', id=5)
(cast(dt as TEXT)='2015-02-23T080000.246000 UTC', id=4)
(cast(dt as TEXT)='2015-02-23T080000.246246 UTC', id=6)
(cast(dt as TEXT)='2015-02-23T080000.246000 UTC', id=4)
(cast(dt as TEXT)='2015-02-23T080000.000246 UTC', id=5)
(cast(dt as TEXT)='2015-02-23T080000.246246 UTC', id=6)
(cast(dt as TEXT)='1969-12-31T200000.246000 US/Eastern', id=1)
(cast(dt as TEXT)='1969-12-31T200000.246246 US/Eastern', id=2)
(cast(dt as TEXT)='1969-12-31T210000.246000 US/Eastern', id=3)
(cast(dt as TEXT)='2015-02-23T030000.000246 US/Eastern', id=5)
(cast(dt as TEXT)='2015-02-23T030000.246000 US/Eastern', id=4)
(cast(dt as TEXT)='2015-02-23T030000.246246 US/Eastern', id=6)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(rows inserted=1)
(cast(ds as TEXT)=NULL, id=1)
(cast(ds as TEXT)=NULL, id=2)
(cast(ds as TEXT)=NULL, id=3)
(cast(ds as TEXT)=NULL, id=4)
(cast(ds as TEXT)=NULL, id=5)
(cast(ds as TEXT)=NULL, id=6)
(cast(ds as TEXT)='0 00:00:01.246000', id=10)
(cast(ds as TEXT)='0 00:00:01.246246', id=11)
(cast(ds as TEXT)='0 00:00:10.000000', id=12)
(cast(ds as TEXT)='0 01:00:00.246000', id=7)
(cast(ds as TEXT)='0 01:00:00.246246', id=8)
(cast(ds as TEXT)='0 02:00:00.246000', id=9)
(cast(ds as TEXT)='0 02:04:06.246000', id=14)
(cast(ds as TEXT)='10 00:00:00.123456', id=13)
(cast(ds as TEXT)='0 01:00:00.246000', id=7)
(cast(ds as TEXT)='0 01:00:00.246246', id=8)
(cast(ds as TEXT)='0 02:00:00.246000', id=9)
(cast(ds as TEXT)='0 02:04:06.246000', id=14)
(cast(ds as TEXT)='10 00:00:00.123456', id=13)
(cast(dt as TEXT)='1969-12-31T200000.246000 US/Eastern', cast(ds as TEXT)=NULL, id=1)
(cast(dt as TEXT)='1969-12-31T200000.246246 US/Eastern', cast(ds as TEXT)=NULL, id=2)
(cast(dt as TEXT)='1969-12-31T210000.246000 US/Eastern', cast(ds as TEXT)=NULL, id=3)
(cast(dt as TEXT)='2015-02-23T030000.246000 US/Eastern', cast(ds as TEXT)=NULL, id=4)
(cast(dt as TEXT)='2015-02-23T030000.000246 US/Eastern', cast(ds as TEXT)=NULL, id=5)
(cast(dt as TEXT)='2015-02-23T030000.246246 US/Eastern', cast(ds as TEXT)=NULL, id=6)
(cast(dt as TEXT)=NULL, cast(ds as TEXT)='0 01:00:00.246000', id=7)
(cast(dt as TEXT)=NULL, cast(ds as TEXT)='0 01:00:00.246246', id=8)
(cast(dt as TEXT)=NULL, cast(ds as TEXT)='0 02:00:00.246000', id=9)
(cast(dt as TEXT)=NULL, cast(ds as TEXT)='0 00:00:01.246000', id=10)
(cast(dt as TEXT)=NULL, cast(ds as TEXT)='0 00:00:01.246246', id=11)
(cast(dt as TEXT)=NULL, cast(ds as TEXT)='0 00:00:10.000000', id=12)
(cast(dt as TEXT)=NULL, cast(ds as TEXT)='10 00:00:00.123456', id=13)
(cast(dt as TEXT)=NULL, cast(ds as TEXT)='0 02:04:06.246000', id=14)
(count(*)=6)
